src/main/java/org/folio/oaipmh/helpers/records/RecordMetadataManager.java

```diff
@@ -209,35 +241,56 @@ private void addSubFieldGroup(Map<String, Object> effectiveLocationSubFields, Js
    *
    * @param metadataSource      - record source
    * @param metadataSourceOwner - record source owner
-   * @param request             - OAI-PMH request
    * @return record source
    */
-  public String updateMetadataSourceWithDiscoverySuppressedDataIfNecessary(String metadataSource, JsonObject metadataSourceOwner,
-      Request request) {
-    if (getBooleanProperty(request.getOkapiHeaders(), REPOSITORY_SUPPRESSED_RECORDS_PROCESSING)) {
+  public String updateMetadataSourceWithDiscoverySuppressedData(String metadataSource, JsonObject metadataSourceOwner) {
       JsonObject content = new JsonObject(metadataSource);
       JsonArray fields = content.getJsonArray(FIELDS);
       Optional<JsonObject> generalInfoDataFieldOptional = getGeneralInfoDataField(fields);
-      if (generalInfoDataFieldOptional.isPresent()) {
-        updateDataFieldWithDiscoverySuppressedData(generalInfoDataFieldOptional.get(), metadataSourceOwner);
-      } else {
-        appendGeneralInfoDatafieldWithDiscoverySuppressedData(fields, metadataSourceOwner);
-      }
+      generalInfoDataFieldOptional.ifPresent(jsonObject ->
+      updateDataFieldWithDiscoverySuppressedData(jsonObject, metadataSourceOwner, GENERAL_INFO_FIELD_TAG_NUMBER));
       return content.encode();
-    }
-    return metadataSource;
   }
 
+  /**
+   * Updates marc electronic access field(tag=856, ind1=ind2='f') with additional subfield which holds data about record discovery
+   * suppression status. Additional subfield has code = 't' and value = '0' if record is discovery suppressed and '1' at opposite
+   * case.
+   *
+   * @param metadataSource      - record source
+   * @param metadataSourceOwner - record source owner
+   * @return record source
+   */
+  public String updateElectronicAccessFieldWithDiscoverySuppressedData(String metadataSource, JsonObject metadataSourceOwner) {
+    JsonObject content = new JsonObject(metadataSource);
+    JsonArray fields = content.getJsonArray(FIELDS);
+    Optional<JsonObject> electronicAccessField = getElectronicAccessField(fields);
+    electronicAccessField.ifPresent(jsonObject ->
+      updateDataFieldWithDiscoverySuppressedData(jsonObject, metadataSourceOwner, ELECTRONIC_ACCESS_FILED_TAG_NUMBER));
+    return content.encode();
+  }
+
+
   private Optional<JsonObject> getGeneralInfoDataField(JsonArray fields) {
     return fields.stream()
       .map(obj -> (JsonObject) obj)
       .filter(generalInfoFieldPredicate)
       .findFirst();
   }
 
+  private Optional<JsonObject> getElectronicAccessField(JsonArray fields) {
```

Comment:
Taking into account that JsaonArray is raw typed, is it fully safe to use implicit type casting here? Possibly it would be better to do slemting like iteration over stream range and getting JsonObject by index: 

IntStream.range(0, fields.size()).mapToObj(idx -> fields.getJsonObject(idx)).filter(electronicAccessPredicate).findFirst();

Reply:
I think it is overcomplication, we'll have to change everywhere else.